pkg/controller/pxc/users.go

```diff
@@ -354,54 +355,55 @@ func (r *ReconcilePerconaXtraDBCluster) manageSysUsers(cr *api.PerconaXtraDBClus
 }
 
 func (r *ReconcilePerconaXtraDBCluster) syncPXCUsersWithProxySQL(cr *api.PerconaXtraDBCluster) error {
-	if cr.Status.Status != api.AppStateReady || cr.Status.ProxySQL.Status != api.AppStateReady {
+	if cr.Spec.ProxySQL == nil || !cr.Spec.ProxySQL.Enabled || cr.Status.PXC.Ready < 1 {
 		return nil
 	}
-	// sync users if ProxySql enabled
-	if cr.Spec.ProxySQL == nil || !cr.Spec.ProxySQL.Enabled || cr.Status.ObservedGeneration != cr.Generation || cr.Status.PXC.Ready < 1 {
+	if cr.Status.Status != api.AppStateReady || cr.Status.ProxySQL.Status != api.AppStateReady {
 		return nil
 	}
-	pod := corev1.Pod{}
-	err := r.client.Get(context.TODO(),
-		types.NamespacedName{
-			Namespace: cr.Namespace,
-			Name:      cr.Name + "-proxysql-0",
-		},
-		&pod,
-	)
-	if err != nil && k8serrors.IsNotFound(err) {
-		return err
-	} else if err != nil {
-		return errors.Wrap(err, "get proxysql pod")
-	}
-	var errb, outb bytes.Buffer
-	err = r.clientcmd.Exec(&pod, "proxysql", []string{"proxysql-admin", "--syncusers"}, nil, &outb, &errb, false)
-	if err != nil {
-		return errors.Errorf("exec syncusers: %v / %s / %s", err, outb.String(), errb.String())
-	}
-	if len(errb.Bytes()) > 0 {
-		return errors.New("syncusers: " + errb.String())
+
+	for i := 0; i < int(cr.Spec.PXC.Size); i++ {
+		pod := corev1.Pod{}
+		err := r.client.Get(context.TODO(),
+			types.NamespacedName{
+				Namespace: cr.Namespace,
+				Name:      cr.Name + "-proxysql-" + strconv.Itoa(i),
```

Comment:
will it work if we have different pxc and proxysql size?

Reply:
fixed https://github.com/percona/percona-xtradb-cluster-operator/pull/795/commits/6e5e4509b1cb4ab8ff070822cc598f8ea3f8e08a